lib/routers/profile.js

```diff
@@ -9,13 +9,22 @@ router.param('id', (req, res, next, id) => {
     return next(new NotFoundError());
   }
   const { Profile } = req.models;
+
+  const profile = Profile.scopeSingle({
+    id: req.params.id,
+    establishmentId: req.establishment.id,
+    userId: req.profile.id
+  });
+
   Promise.resolve()
-    .then(() => {
-      return Profile.query()
-        .findById(req.params.id)
-        .where('establishments.id', req.establishment.id)
-        .joinRelation('establishments')
-        .eager('[roles.places, establishments, pil, projects, trainingModules]');
+    .then(() => req.user.can('profile.read.all', req.params))
+    .then(allowed => {
+      if (allowed) {
+        return profile.get();
+      }
+      return Promise.resolve()
+        .then(() => req.user.can('profile.read.basic', req.params))
+        .then(allowed => allowed ? profile.getNamed() : next(new Error('Not found')));
```

Comment:
You'll want to throw the error rather than call `next`, because calling `next` will allow the promise chain to keep resolving its `then`s, which I don't think you want.

Reply:
yarp :)

Reply:
Also, throw a `NotFoundError` - https://github.com/UKHomeOffice/asl-public-api/blob/master/lib/errors/not-found.js and it'll handle status codes and shizzle for you.